tools/modules/update.sh

```diff
@@ -22,21 +22,24 @@
 # repositories are at the correct version and will configure and build them.
 
 wget_copy_archive() {
+  # The dependency to download (llvm, clang).
   local target="$1"
-  local dir="$2"
+  # The directory to download the dependency into.
+  local dir="${2:?missing directory}"
+  # The specific version of the dependency to use.
   local sha="$3"
 
   if [[ ! -f "$dir/$sha.sentinel" ]]; then
     wget "https://github.com/llvm-mirror/$target/archive/$sha.zip"
+    trap 'rm "$sha.zip"' EXIT ERR INT
     rm -rf "$dir"
     mkdir -p "$dir"
     local tmpdir=$(mktemp -d)
+    trap 'rm -rf "$tmpdir"' EXIT ERR INT
```

Comment:
I believe `EXIT` subsumes the others.

Reply:
I believe it depends on the shell implementation, so I always just play it safe.

Reply:
Sure I am fine having ERR INT in there as well